base/c.jl

```diff
@@ -43,6 +43,29 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{Cchar} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+immutable Cstring; p::Ptr{Cchar}; end
+immutable Cwstring; p::Ptr{Cwchar_t}; end
```

Comment:
Okay,
```jl
const sizeof_Int = box(UInt32, unbox(Int32, Int.size))
if sizeof_Int === 0x00000008 # need === since == doesn't exist yet              
    bitstype 64 Cstring
    bitstype 64 Cwstring
else
    bitstype 32 Cstring
    bitstype 32 Cwstring
end
```
seems to work, though why I need the `const` declaration to avoid `type definition not allowed inside a local scope` is mysterious to me.